public/js/components/FormFields/FormFieldDateTextInput.js

```diff
@@ -0,0 +1,58 @@
+import React, {PropTypes} from 'react';
+import ShowErrors from '../Utilities/ShowErrors';
+import { errorPropType } from '../../constants/errorPropType';
+import moment from 'moment';
+
+export default class FormFieldDateTextInput extends React.Component {
+
+
+    static propTypes = {
+        fieldLabel: PropTypes.string,
+        fieldName: PropTypes.string,
+        fieldValue: PropTypes.number,
+        fieldErrors: PropTypes.arrayOf(errorPropType),
+        formRowClass: PropTypes.string,
+        onUpdateField: PropTypes.func
+    };
+
+    milisecondsToString = (ms) => {
+        moment(ms).format("yyyy-MM-dd");
+    }
+
+    state = {
```

Comment:
whoa 😮  I have never seen this construct; is it legal? Usually `state` is initialised in the constructor, as the constructor is the starting point for using `props`.

Reply:
Yup and we had to do it this way otherwise it breaks 😱  this has been an _eye-opening_ PR....

Reply:
it's certainly used elsewhere in atom-workshop!

Reply:
so much for React and JavaScript classes 😝  well done 👍

Reply:
@regiskuckaertz - Only because we're using https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-class-properties